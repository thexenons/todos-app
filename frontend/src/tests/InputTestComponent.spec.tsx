import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import InputTestComponent from "./InputTestComponent";

describe("InputTestComponent", () => {
	it("Render", () => {
		const { container } = render(<InputTestComponent />);

		expect(container).toMatchSnapshot();
	});

	it("submit", async () => {
		render(
			<InputTestComponent>
				<button type="submit">Submit</button>
			</InputTestComponent>
		);

		const button = screen.getByRole("button");
		expect(button).toBeInTheDocument();

		await userEvent.click(button);
	});
});
