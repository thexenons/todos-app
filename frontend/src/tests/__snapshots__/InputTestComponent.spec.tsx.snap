// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputTestComponent > Render 1`] = `
<div>
  <form />
</div>
`;
